src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/CSharpFormattingOptions2.cs

```diff
@@ -164,17 +160,42 @@ private static Option2<bool> CreateNewLineForBracesOption(NewLineOption newLineO
             EditorConfigStorageLocation.ForBoolOption("csharp_space_after_keywords_in_control_flow_statements"),
             new RoamingProfileStorageLocation("TextEditor.CSharp.Specific.SpaceAfterControlFlowStatementKeyword"));
 
-        public static Option2<bool> SpaceWithinExpressionParentheses { get; } = CreateSpaceWithinParenthesesOption(
-            SpacingWithinParenthesesOption.Expressions, nameof(SpaceWithinExpressionParentheses),
-            CSharpSyntaxFormattingOptions.SpacingDefault.HasFlag(SpacePlacement.WithinExpressionParentheses));
-
-        public static Option2<bool> SpaceWithinCastParentheses { get; } = CreateSpaceWithinParenthesesOption(
-            SpacingWithinParenthesesOption.TypeCasts, nameof(SpaceWithinCastParentheses),
-            CSharpSyntaxFormattingOptions.SpacingDefault.HasFlag(SpacePlacement.WithinCastParentheses));
-
-        public static Option2<bool> SpaceWithinOtherParentheses { get; } = CreateSpaceWithinParenthesesOption(
-            SpacingWithinParenthesesOption.ControlFlowStatements, nameof(SpaceWithinOtherParentheses),
-            CSharpSyntaxFormattingOptions.SpacingDefault.HasFlag(SpacePlacement.WithinOtherParentheses));
+        // legacy option:
+        public static Option2<bool> SpaceWithinExpressionParentheses { get; } = new Option2<bool>(
+            feature: "CSharpFormattingOptions",
+            CSharpFormattingOptionGroups.Spacing,
+            name: "SpaceWithinExpressionParentheses",
+            defaultValue: CSharpSyntaxFormattingOptions.SpacingDefault.HasFlag(SpacePlacement.WithinExpressionParentheses),
+            new RoamingProfileStorageLocation("TextEditor.CSharp.Specific.SpaceWithinExpressionParentheses"));
```

Comment:
This feels like a breaking change for the public API:

https://github.com/dotnet/roslyn/blob/19704e6876c4a5a88267af5cf83a4bfe6287d356/src/Workspaces/CSharp/Portable/Formatting/CSharpFormattingOptions.cs#L34

Reply:
What I meant here is if someone wants to read this option, it will now be read directly from roaming storage instead of checking editorconfig first.

Reply:
What option reading API do you have on mind?